File: app/src/main/java/com/jdappel/beerinvestigator/rest/BreweryDBApi.java
package com.jdappel.beerinvestigator.rest;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Defines the API endpoints that are used with {@code Retrofit} to access BreweryDB
 */
public interface BreweryDBApi {

    @GET("http://api.brewerydb.com/v2/search?type=beer")
    Observable<BreweryDBResponse<Beer>> getBeers(@Query("q") String query);
}

File: app/src/main/java/com/jdappel/beerinvestigator/rest/BreweryDBApi.kt
package com.jdappel.beerinvestigator.rest

import retrofit2.http.GET
import com.jdappel.beerinvestigator.model.Beer
import retrofit2.Response
import retrofit2.http.Query

/**
 * Defines the API endpoints that are used with `Retrofit` to access BreweryDB
 */
interface BreweryDBApi {
    @GET("http://api.brewerydb.com/v2/search?type=beer")
    suspend fun getBeers(@Query("q") query: String?): Response<BreweryDBResponse<Beer>>
}